MobileApplication/MobileApplication/ViewModels/Transaction/ItemsViewModel.cs

```diff
@@ -27,22 +33,34 @@ public ItemsViewModel()
 
             Title = "Transactions";
             Items = new ObservableCollection<Transaction>();
+            AllItems = new ObservableCollection<Transaction>();
 
             LoadItemsCommand = new Command(async () => await ExecuteLoadItemsCommand());
             ItemTapped = new Command<Transaction>(OnItemSelected);
             AddItemCommand = new Command(OnAddItem);
+
+            DateTo = DateTime.Now;
+
+            FilterCommand = new Command(OnFilter, ValidateFilter);
+            ResetFilterCommand = new Command(OnResetFilter);
+            this.PropertyChanged +=
+                (_, __) => FilterCommand.ChangeCanExecute();
         }
 
         async Task ExecuteLoadItemsCommand()
         {
             IsBusy = true;
+            AllItems.Clear();
             Items.Clear();
 
             try
             {
                 var items = await RestService.RefreshDataAsync();
 
+                items.ForEach(transaction => AllItems.Add(transaction));
                 items.ForEach(transaction => Items.Add(transaction));
```

Comment:
Even if you need two lists, why are you performing insertions twice instead of duplicating the existing one when needed?